### DISCUSSION TITLE
Sanitize HTML on ServerSide

### DISCUSSION THREAD

**Discussion by KSK-Exipnos** [wp-graphql/wp-graphql#2845] [Q&A]
I'm working with WpGraphQl and a decoupled Vue3 frontend. 
To display the posts content, I'm using v-html="". 

With planned post creation by the user on the front end, the HTML isn't sanitized in any way yet.

For sure, I could use a package like "vue-3-sanitize" to clean the output. But doing it on the server side before saving to the db would be better.

Is there already a way to do that with the plugin, or do I need to add a middleware.
And if so, some hints would be appreciated.

**Comment by justlevine** [comment]
Hey @KSK-Exipnos 

1. WordPress by default sanitizes the `post_content` when it is saved to the database.
2. WordPress by default does _not_ escape the `post_content` when it is retrieved, and WPGraphQL follows that same pattern. The reason for this is that there are _many_ uses for the `content` that go beyond blindly outputting the returned html.
3. If you're worried about a situation where your _database itself_ is compromised (vs an unauthorized user with elevated permissions), you can use `add_filter( 'the_content', 'wp_kses_post' );` or a callback method on `the_content` that calls `wp_kses_post()` conditionally (e.g. only if it's a WPGraphQL request)
